streamalert/shared/config.py

```diff
@@ -218,9 +218,37 @@ def load_config(conf_dir='conf/', exclude=None, include=None, validate=True):
     if validate:
         _validate_config(config)
 
+    if config.get('logs'):
+        config['logs'] = _sanitize_logs_name(config['logs'])
+
+    if (config.get('global')
+            and config['global'].get('infrastructure')
+            and config['global']['infrastructure'].get('firehose')
+            and config['global']['infrastructure']['firehose'].get('enabled_logs')):
+        config['global']['infrastructure']['firehose']['enabled_logs'] = _sanitize_logs_name(
+            config['global']['infrastructure']['firehose']['enabled_logs']
+        )
+
     return config
 
 
+def _sanitize_logs_name(config):
+    """Sanitize the name of logs and replace the dots with underscores. For some reason,
+    we have log name with dots in it in the conf/schemas/carbonblack.json or conf/logs.json.
+
+    Args:
+        config (dict): loaded config from conf/ directory
+
+    Returns:
+        new_config (dict): new config with sanitized keys
+    """
+    new_config = dict()
+    for key, _ in config.items():
+        sanitized_key = key.replace('.', '_')
```

Comment:
I think we should use that same regex here to do special char replacement: 
https://github.com/airbnb/streamalert/blob/319fc115d3cf42dea6df10f3a9d9ed7583671f7f/streamalert/classifier/clients/firehose.py#L299

Since that finds all *non-word* characters (except for the `_` which it allows):

https://www.rexegg.com/regex-quickstart.html

Reply:
I would, `re.sub(cls.SPECIAL_CHAR_REGEX, cls.SPECIAL_CHAR_SUB, log_name)` was my first fix. But we have to `:` because firehose use the `:` to determine if we enable a set of logs or single log, e.g. `osquery` vs. `osquery:differential`.

Reply:
didn't we used to split on the `:` and the only worry about sanitizing the second half? everything _after_ the `:`, then concatenating that with the first half again?

Reply:
I worry cause someone _might_ use a different character that will also break this... like `-` ...

Reply:
Aha, actually this is a great idea!!! Smart smart...